tests/cli/test_list.py

```diff
@@ -0,0 +1,172 @@
+    * (at your option) any later version.
+    *
+    * This program is distributed in the hope that it will be useful,
+    * but WITHOUT ANY WARRANTY; without even the implied warranty of
+    * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    * GNU General Public License for more details.
+    *
+    * You should have received a copy of the GNU General Public License
+    * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+    -->
+"""
+
+import unittest
+from unittest.mock import patch, call
+from scribe_data.cli.list import (
+    list_languages,
+    list_data_types,
+    list_all,
+    list_languages_for_data_type,
+    list_wrapper,
+)
+from scribe_data.cli.main import main
+
+class TestListFunctions(unittest.TestCase):
+    @patch("builtins.print")
+    def test_list_languages(self, mock_print):
+        list_languages()
+        expected_calls = [
+            call(),
+            call('Language     ISO  QID    '),
+            call('-----------------------'),
+            call('English      en   Q1860  '),
+            call('French       fr   Q150   '),
+            call('German       de   Q188   '),
+            call('Italian      it   Q652   '),
+            call('Portuguese   pt   Q5146  '),
+            call('Russian      ru   Q7737  '),
+            call('Spanish      es   Q1321  '),
+            call('Swedish      sv   Q9027  '),
+            call('-----------------------'),
+            call(),
+        ]
+        mock_print.assert_has_calls(expected_calls)
+
+    @patch("builtins.print")
+    def test_list_data_types_all_languages(self, mock_print):
+        list_data_types()
+        expected_calls = [
+            call(),
+            call('Available data types: All languages'),
+            call('-----------------------------------'),
+            call('emoji-keywords'),
+            call('nouns'),
+            call('prepositions'),
+            call('translations'),
+            call('verbs'),
+            call('-----------------------------------'),
+            call(),
+        ]
+        mock_print.assert_has_calls(expected_calls)
+
+    @patch("builtins.print")
+    def test_list_data_types_specific_language(self, mock_print):
+        list_data_types("English")
+        expected_calls = [
+            call(),
+            call('Available data types: English'),
+            call('-----------------------------'),
+            call('emoji-keywords'),
+            call('nouns'),
+            call('translations'),
+            call('verbs'),
+            call('-----------------------------'),
+            call(),
+        ]
+        mock_print.assert_has_calls(expected_calls)
+
+    def test_list_data_types_invalid_language(self):
+        with self.assertRaises(ValueError):
+            list_data_types("InvalidLanguage")
+
+    def test_list_data_types_no_data_types(self):
+        with self.assertRaises(ValueError):
+            list_data_types("Klingon")  
```

Comment:
Maybe one day could be a language 😅 Wikidata does have data for it 😇 But fine for now :)